### PR TITLE
Refactor/profile

### PR DESCRIPTION
**Pull request by ahmadtirmazi**

### CODE REVIEW

File: src/endpoints/json-rpc-provider.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { EosRpcSource, GetTableRowsOptions } from '@alien-worlds/api-core';
+
+import { GetTableRowsResult } from 'eosjs/dist/eosjs-rpc-interfaces';
+
+export class JsonRpcProvider extends EosRpcSource {
```
Comment (rkamysz) [inline src/endpoints/json-rpc-provider.ts]: I think we dont need this file anymore as we have EosJsonRpc in api-core and if you want to keep it I would move this fiel to some common or inside ioc.config as this is clear data source

File: src/endpoints/profile/domain/entities/profile-item.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  /**
+   * Create DTO based on entity data
+   *
+   * @returns {ProfileItemDocument}
+   */
+  public toDto(): ProfileItemDocument {
+    const dto = {
+      description: this.description,
+      email: this.email,
+      familyName: this.familyName,
```
Comment (rkamysz) [inline src/endpoints/profile/domain/entities/profile-item.ts:33]: do we have camel case in mongodb documents?.... shouldn't it be family_name etc.

File: src/endpoints/profile/domain/profile.controller.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		private isProfileFlaggedUseCase: IsProfileFlaggedUseCase
+	) { }
+
+	/*methods*/
+
+	/**
+	 *
+	 * @returns {Promise<Result<ProfileOutput, Error>>}
+	 */
+	public async profile(
```
Comment (rkamysz) [inline src/endpoints/profile/domain/profile.controller.ts:41]: It's ok for now, but keep in mind that if you had more methods in that controller and each of those methods contained logic, such a controller would be huge. the controller should be more like a command/use case dispatcher

File: src/endpoints/profile/domain/use-cases/is-profile-flagged.use-case.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      })
+
+      const result = await this.flagRepository.aggregate(queryModel);
+      if (result.isFailure) {
+        return Result.withFailure(result.failure)
+      }
+
+      output = result.content.map(doc => {
+        return {
+          account: doc.cand,
```
Comment (rkamysz) [inline src/endpoints/profile/domain/use-cases/is-profile-flagged.use-case.ts]: we should use full names in entities

**Review by rkamysz — CHANGES_REQUESTED** [review]

**Comment by rkamysz** [comment]
looks good .... just names, you dont have to touch controller

File: src/endpoints/profile/domain/entities/profile-item.ts (hunk 2 above)
Comment (ahmadtirmazi) [reply src/endpoints/profile/domain/entities/profile-item.ts:33]: Yes, in this case we already have camel case in mongo db.

**Review by rkamysz — APPROVED** [review]
LGTM thank you

**Pull request merged**